data/utils/defectsFromBias.py

```diff
@@ -21,104 +21,43 @@
-    """Write a defects FITS file.
-
-    Parameters
-    ----------
-    bboxList : `list` of `lsst.geom.Box2I`
-        List of bounding boxes defining defect locations.
-    path : `str`
-        Path of output defects file; should end with ".fits".
-    detectorSerial : `str`
-        Serial code of detector.
-    detectorName : `str`
-        Name of detector.
-    """
-    head = fits.Header()
-    head.update('SERIAL', detectorSerial, 'Serial of the detector')
-    head.update('NAME', detectorName, 'Name of detector for this defect map')
-    head.update('CDATE', time.asctime(time.gmtime()), 'UTC of creation')
-
-    x0 = numpy.array([d.getMinX() for d in bboxList])
-    y0 = numpy.array([d.getMinY() for d in bboxList])
-    width = numpy.array([d.getWidth() for d in bboxList])
-    height = numpy.array([d.getHeight() for d in bboxList])
-
-    col1 = fits.Column(name='x0', format='I', array=numpy.array(x0))
-    col2 = fits.Column(name='y0', format='I', array=numpy.array(y0))
-    col3 = fits.Column(name='height', format='I', array=numpy.array(height))
-    col4 = fits.Column(name='width', format='I', array=numpy.array(width))
-    cols = fits.ColDefs([col1, col2, col3, col4])
-    tbhdu = fits.new_table(cols, header=head)
-    hdu = fits.PrimaryHDU()
-    thdulist = fits.HDUList([hdu, tbhdu])
-    thdulist.writeto(DefectsPath)
-
-
 if __name__ == "__main__":
     parser = argparse.ArgumentParser(
-        description="""Construct a defects file from the mask plane of a test camera bias frame.
+        description=f"""Construct a defects file from the mask plane of a test camera bias frame.
 To use this command you must setup ip_isr and astropy.
-Output is written to the current directory as file %r, which must not already exist.
-""" % (DefectsPath,)
+Output is written to the current directory as file {DefectsPath}, which must not already exist.
+"""
     )
     parser.add_argument("bias", help="path to bias image for the test camera")
     args = parser.parse_args()
 
     biasMI = afwImage.MaskedImageF(args.bias)
-    defectList = getDefectListFromMask(biasMI, "BAD")
-    bboxList = [defect.getBBox() for defect in defectList]
-    writeDefectsFile(bboxList, DefectsPath, detectorSerial, detectorName)
+    defectList = Defects.fromMask(biasMI, "BAD")
+    valid_start = dateutil.parser.parse('19700101T000000')
+    md = defectList.getMetadata()
+    md['INSTRUME'] = 'test'
+    md['DETECTOR'] = detectorName
+    md['CALIBDATE'] = valid_start.isoformat()
+    md['FILTER'] = None
+    defect_file = defectList.writeText(DefectsPath)
     print("wrote defects file %r" % (DefectsPath,))
 
-    test2BBoxList = getBBoxList(DefectsPath, detectorName)
-    assert len(bboxList) == len(test2BBoxList)
-    for boxA, boxB in zip(bboxList, test2BBoxList):
-        assert boxA == boxB
+    test2defectList = Defects.readText(defect_file)
+    assert defectList == test2defectList
+    for dA, dB in zip(defectList, test2defectList):
```

Comment:
This second for loop doesn't do anything that the `==` line before doesn't do. This is how `Defects` testing for equality is done.

Reply:
Oh, of course.  Thanks for pointing that out.